### DISCUSSION TITLE
Best way to handle empty bodies in error response?

### DISCUSSION THREAD

**Discussion by justin-harr** [sindresorhus/ky#438] [Q&A]
Currently I use KY like this to fetch my data and get either the success or error json:
```
  const handleGetUser = async () => {
    try {
      const { user } = await ky.get('/api/user').json();
      console.log(user);
    } catch (error) {
      const { message } = await error.response.json();
      console.log(message || 'Something went wrong');
    }
  };
```

This works great for all "planned errors", but in the case where somehow I hit a route that doesn't exist or the body is empty, `error.response.json()` throws an error. I can solve this by adding another catch like this:

```
const { message } = await error.response.json().catch(() => ({ message: 'Something went wrong' }));
console.log(message || 'Something went wrong');
```
However, that is obviously really ugly. Is there a clean way to handle this? Or at least a way I can move that into a ky hook so that I don't have to replicate that everywhere?

**Comment by Moshyfawn** [comment]
Related? #415
